src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -801,7 +801,7 @@ private int[] ParseLineStarts()
                 var index = 0;
                 if (lastWasCR)
                 {
-                    if (buffer.Length > 0 && buffer[0] == '\n')
+                    if (length > 0 && buffer[0] == '\n')
```

Comment:
This change makes me a bit nervous.  This seems like it's been wrong for a very long time.  How often is this code path hit?